### ISSUE TITLE
"stack upgrade --git" rebuilds even when stack if up-to-date

### ISSUE THREAD

**Issue by rvion** [commercialhaskell/stack#811]
`stack upgrade --git` rebuilds stack even when stack is up-to-date
To repro, one can run `stack upgrade --git` twice and see what happens

I have
```
stack --version
Version 0.1.3.1, Git revision 3da2a5b1faf8ef045fed550840647571cda9ff3e
```

**Comment by snoyberg** [comment]
This is enough of a corner case that I wasn't worried about it. If you'd like to improve the situation, PR welcome, all of the code should be in Stack.Upgrade.

**Comment by rvion** [comment]
works for me, I'll give it a try once I'll have taken the time to finish setting up appveyor

**Comment by snoyberg** [comment]
Resolved by #937
